lib/proc/proc_snippet.c

```diff
@@ -250,20 +260,47 @@ func_snippet_html(grn_ctx *ctx, int nargs, grn_obj **args,
 {
   grn_obj *snippets = NULL;
 
-  /* TODO: support parameters */
-  if (nargs == 1) {
+  if (nargs > 0) {
     grn_obj *text = args[0];
     grn_obj *expression = NULL;
     grn_obj *condition_ptr = NULL;
     grn_obj *condition = NULL;
     grn_obj *snip = NULL;
+    grn_obj *default_return_value = NULL;
     int flags = GRN_SNIP_SKIP_LEADING_SPACES;
     unsigned int width = 200;
     unsigned int max_n_results = 3;
     const char *open_tag = "<span class=\"keyword\">";
     const char *close_tag = "</span>";
     grn_snip_mapping *mapping = GRN_SNIP_MAPPING_HTML_ESCAPE;
 
+    if (args[1] && args[1]->header.type == GRN_TABLE_HASH_KEY) {
+      grn_obj *options = args[1];
+      void *key;
+      int key_size;
+      grn_obj *value;
+  
+      GRN_HASH_EACH_BEGIN(ctx, (grn_hash *)options, cursor, id) {
+        grn_raw_string option_name;
+        option_name.value = "default";
+        option_name.length = strlen("default");
+
+        grn_hash_cursor_get_key_value(ctx, cursor,
+                                      &key, &key_size,
+                                      (void **)&value);
+        if (key_size == option_name.length
+            && !memcmp(key, option_name.value, option_name.length)) {
+          default_return_value = value;
+        } else {
+          GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                           "snippet_html(): invalid option name: <%.*s>",
+                           key_size, (char *)key);
+          grn_hash_cursor_close(ctx, cursor);
+          goto exit;
```

Comment:
We should not use `goto` in `GRN_HASH_EACH_BEGIN {}`.
We should `break` and `goto`.